src/swap/core/useGetTokenBalance.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react';
+import { useReadContract } from 'wagmi';
+import { erc20Abi, formatUnits } from 'viem';
+import { getRoundedAmount } from '../../utils/getRoundedAmount';
+import type { Address } from 'viem';
+import type { Token } from '../../token';
+import type { UseReadContractReturnType } from 'wagmi';
+
+export function useGetTokenBalance(address: Address, token?: Token) {
```

Comment:
Same, let's have this in Wallet.

I see Wallet as accessing all info related to a particular Account.

Reply:
Btw I really like the simple API you wrote as `useGetTokenBalance(address: Address, token?: Token)`, very nice.

Btw why `token` is optional?

Reply:
`token` is optional because `token` and `swappableTokens` are optional props for `SwapAmountInput` so there is a chance that `token` is undefined in `Swap` component ( typescript gets upset )

Reply:
Ok so, small question. Does `useGetTokenBalance` works without passing a Token?

If not, we should make it mandatory, if does, than optional is fine.

Ideally the Hooks type should reflect how the Hook works, and than if in case in Swap we don't have a Token, than we use conditinals to cover that case in the Swap component.

Reply:
it does not work without a `token` (it will be disabled). since it is a hook, i don't think we are able to conditionally call it. do you have another work around in mind ?